alchemist/alchemist-incarnation-biochemistry/src/test/java/it/unibo/alchemist/test/TestBioRect2DEnvironmentNoOverlap.java

```diff
@@ -39,12 +31,14 @@
 import it.unibo.alchemist.model.interfaces.Reaction;
 import it.unibo.alchemist.model.interfaces.Time;
 
+import static org.junit.Assert.*;
+
 /**
  *
  */
 public class TestBioRect2DEnvironmentNoOverlap {
 
-    private static final double MAX_PRECISION = 0.9999999999;
+    private static final double DELTA = 1e-6;
```

Comment:
Looks rather arbitrary: up to which value does the test work? What happens if you put a zero? What if you put `nextUp(0)`?

Reply:
The test works for values up to DELTA=1e-7 and starts failing with DELTA=1e-8 (same for DELTA=0, obviously, but giving in the error log that two nodes with distance d=1.0 are overlapping, which should be acceptable so a DELTA is needed).
Eventually, as nextUp(0) returns a value which is about 1e-45, the test wil fail too.

Reply:
I'd say here documenting (with a comment in code) which is the limit at which the test works is fine.

Reply:
Now it should be ok